### PR TITLE
fix(policies): Re-revert the policies fix + ingest documents directly to search

### PR DESCRIPTION
**Pull request by dexter-mh-lee**
The initial PR https://github.com/datahub-project/datahub/pull/4713 failed when run on kubernetes. Reason was that the MCLs produced by the ingestPoliciesStep to fill up the policies index was being consumed by the already live gms pod (which does not have the new model with searchable fields), so they all get lost. 

In this version, we directly push into search without going through mcl consumer to mitigate this issue. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java [hunk 1]
```diff
@@ -198,15 +199,13 @@ private void updateSearchService(String entityName, Urn urn, AspectSpec aspectSp
       return;
     }
 
-    String docId;
-    try {
-      docId = URLEncoder.encode(urn.toString(), "UTF-8");
-    } catch (UnsupportedEncodingException e) {
-      log.error("Failed to encode the urn with error: {}", e.toString());
+    Optional<String> docId = SearchUtils.getDocId(urn);
```
Comment (jjoyce0510) [inline metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java:202]: Why wouldn't doc id be present? bad urn?

Comment (dexter-mh-lee) [reply metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/hook/UpdateIndicesHook.java:202]: I am suspecting some character that cannot be url encoded. The urlencoder throws this exception

File: metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestPoliciesStep.java [hunk 2]
```diff
@@ -88,11 +102,63 @@ public void execute() throws IOException, URISyntaxException {
     log.info("Successfully ingested default access policies.");
   }
 
+  /**
+   * Send MCLs for each policy to refill the policy search index
```
Comment (jjoyce0510) [inline metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestPoliciesStep.java:106]: minor: comment is stale

Comment (dexter-mh-lee) [reply metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestPoliciesStep.java]: done!

**Review by jjoyce0510 — APPROVED** [review]
overall lgtm

**Pull request merged**